sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy

```diff
@@ -80,6 +80,31 @@ class BlobAPITest extends APISpec {
         stream.toByteArray() == randomData
     }
 
+
+    /* Tests an issue found where buffered upload would not deep copy buffers while determining what upload path to take. */
+    @Unroll
+    def "Upload input stream single upload" () {
+        setup:
+        def randomData = getRandomByteArray(20 * Constants.KB)
+        def input = new ByteArrayInputStream(randomData)
+
+        when:
+        bc.upload(input, 20 * Constants.KB, true)
+
+        then:
+        def stream = new ByteArrayOutputStream()
+        bc.downloadWithResponse(stream, null, null, null, false, null, null)
+        stream.toByteArray() == randomData
+
+        where:
+        size                || _
+        1 * Constants.KB    || _  /* Less than copyToOutputStream buffer size, Less than maxSingleUploadSize */
+        8 * Constants.KB    || _  /* Equal to copyToOutputStream buffer size, Less than maxSingleUploadSize */
+        20 * Constants.KB   || _  /* Greater than copyToOutputStream buffer size, Less than maxSingleUploadSize */
+    }
+
+    /* TODO (gapra): Add more tests to test large data sizes. */
```

Comment:
Thanks for the test. There wasn't anything that could catch that regression.

I'm not sure if adding larger data sizes would bring value in this test.

Reply:
Just a future thought to improve the condition of the upload tests :)